### ISSUE TITLE
release-1.4 testgrid failure

### ISSUE THREAD

**Issue by jessfraz** [kubernetes/kubernetes#43587]
failing: https://k8s-testgrid.appspot.com/release-1.4-blocking#gce-slow-1.4
See: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-slow-release-1.4/3257

Seems to be caused by https://github.com/kubernetes/kubernetes/commit/92427fc1bae65b481c4ea595fabf0e5c034aa6df from https://github.com/kubernetes/kubernetes/pull/42842

pinging @ixdy to see if he has any ideas

**Comment by ixdy** [comment]
sure does look like it. I expected this to mostly be a no-op, so that's frustrating. @MrHohn any ideas?

**Comment by MrHohn** [comment]
Looking, it's sad that addon manager's log is not exposed yet in 1.4 :(

**Comment by MrHohn** [comment]
I'm bringing up a 1.4 cluster for debugging. Will update soon.

**Comment by jessfraz** [comment]
thanks!

**Comment by MrHohn** [comment]
Seems like addon-manager v5.1.1 is somehow broken, logs:
```
== default service account in the kube-system namespace has token default-token-pafjz ==
++ obj /etc/kubernetes/admission-controls/limit-range/limit-range.yaml is created ++
error: name cannot be provided when a selector is specified
error: name cannot be provided when a selector is specified
error: name cannot be provided when a selector is specified
error: name cannot be provided when a selector is specified
error: name cannot be provided when a selector is specified
error: name cannot be provided when a selector is specified
error: name cannot be provided when a selector is specified
error: name cannot be provided when a selector is specified
limitrange "limits" created
...
Error from server: error when creating "/etc/kubernetes/addons/node-problem-detector/node-problem-detector.yaml": daemonsets.extensions "node-problem-detector-v0.1" already exists
INFO: == Failed to execute /usr/local/bin/kubectl  create --namespace=kube-system -f /etc/kubernetes/addons/node-problem-detector/node-problem-detector.yaml at 2017-03-23T21:37:46+0000. 5 tries remaining. ==
Error from server: error when creating "/etc/kubernetes/addons/dashboard/dashboard-service.yaml": services "kubernetes-dashboard" already exists
Error from server: error when creating "/etc/kubernetes/addons/cluster-loadbalancing/glbc/default-svc.yaml": services "default-http-backend" already exists
The Service "kube-dns" is invalid.
spec.clusterIP: Invalid value: "10.0.0.10": provided IP is already allocated
...
```

Looking for the reason.

**Comment by ixdy** [comment]
maybe something did functionally change in kubectl between v1.3.0-beta.2 and v1.3.10?

**Comment by MrHohn** [comment]
> maybe something did functionally change in kubectl between v1.3.0-beta.2 and v1.3.10?

Yeah very likely, let me build another local image and see if kubectl binary is the only issue.

**Comment by MrHohn** [comment]
Oops, built with kubectl v1.3.0-beta.2 and hit the same error:
```
...
WRN: == Kubernetes addon update completed with errors at 2017-03-23T22:10:45+0000 ==
error: name cannot be provided when a selector is specified
...
Error from server: error when creating "/etc/kubernetes/addons/dashboard/dashboard-controller.yaml": replicationcontrollers "kubernetes-dashboard-v1.4.0" already exists
The Service "kube-dns" is invalid.
spec.clusterIP: Invalid value: "10.0.0.10": provided IP is already allocated
...
```

**Comment by MrHohn** [comment]
The addon manager is broken by #30372. Built after reverting this PR and things seem correct.

**Comment by MrHohn** [comment]
Should we re-push addon manager v5.1.1 without this? Should we revert #30372 in 1.4 branch?

**Comment by ixdy** [comment]
oh, good call. I suspect #28823 was also not originally included in v5.1 - it looks like that tag was created in #28467.

**Comment by ixdy** [comment]
Tests started passing again after pushing a new v5.1.1 image. Thanks @jessfraz for alerting us, and thanks @MrHohn for triaging it!
